lm/srilm.py

```diff
@@ -0,0 +1,634 @@
+            "compress",
+            resume="compress",
+            rqmt={"mem": 2, "time": 2, "cpu": 1, "fs": self.fs_rqmt},
+        )
+
+    def create_files(self):
+        """creates bash script for lm creation and compression that will be executed in the run Task"""
+        vocab_str = (
+            f"  -vocab {self.vocab.get_cached_path()} \\\n" if self.vocab is not None else "  -write-vocab vocab \\\n"
+        )
+        multi_kn_str = (
+            f"  -multi-kn-file {self.multi_kn_file.get_path()} \\\n" if self.multi_kn_file is not None else ""
+        )
+        if self.data_mode == ComputeNgramLmJob.DataMode.TEXT:
+            data_str = "-text"
+        elif self.data_mode == ComputeNgramLmJob.DataMode.COUNT:
+            data_str = "-read"
+        else:
+            raise NotImplementedError
+
+        cmd = [
+            f"{self.count_exe.get_path()} \\\n",
+            f"  {data_str} {self.data.get_cached_path()} \\\n",
+            f"{vocab_str}",
+            f"  -order {self.ngram_order} \\\n",
+            f"  -lm ngram.lm \\\n",
+            f"{multi_kn_str}",
+            f"  {' '.join(self.ngram_args)} -unk -memuse\n",
+        ]
+        create_executable("run.sh", cmd)
+        create_executable("compress.sh", [f"gzip -c -9 ngram.lm > ngram.lm.gz\n"])
+
+    def run(self):
+        """executes the previously created lm script and relinks the vocabulary from work folder to output folder"""
+        self.sh("./run.sh")
+        if self.vocab is None:
+            relink("vocab", self.out_vocab.get_path())
+        else:
+            relink(self.vocab.get_path(), self.out_vocab.get_path())
+
+    def compress(self):
+        """executes the previously created compression script and relinks the lm from work folder to output folder"""
+        self.sh("./compress.sh")
+        relink("ngram.lm.gz", self.out_ngram_lm.get_path())
+        if os.path.exists("ngram.lm") and os.path.exists("ngram.lm.gz"):
+            os.remove("ngram.lm")
+
+    @classmethod
+    def hash(cls, kwargs):
+        """delete the queue requirements from the hashing"""
+        del kwargs["mem_rqmt"]
+        del kwargs["cpu_rqmt"]
+        del kwargs["time_rqmt"]
+        del kwargs["fs_rqmt"]
+        return super().hash(kwargs)
+
+
+class ComputeNgramLmPerplexityJob(Job):
+    """
+    Calculate the Perplexity of a Ngram LM via SRILM
+    """
+
+    def __init__(
+        self,
+        ngram_order: int,
+        lm: tk.Path,
+        eval_data: tk.Path,
+        vocab: Optional[tk.Path] = None,
+        set_unknown_flag: bool = True,
+        ppl_args: Optional[str] = None,
+        ngram_exe: Optional[tk.Path] = None,
+        mem_rqmt: int = 16,
+        time_rqmt: float = 12,
+        cpu_rqmt: int = 1,
+        fs_rqmt: str = "10G",
+    ):
+        """
+        :param ngram_order: Maximum n gram order
+        :param lm: LM to evaluate
+        :param vocab: Vocabulary file
+        :param eval_data: Data to calculate PPL on
```

Comment:
```suggestion
        :param eval_data: Data to calculate PPL on
        :param vocab: Vocabulary file
        :param set_unknown_flag: TODO
```
same order as in function definition + missing docstring for `set_unknown_flag`